### ISSUE TITLE
Add Rewrite tool to Jade

### ISSUE THREAD

**Issue by brunobuzzi** [jgfoster/Jade#45]
The original tool description can be found at:
https://refactory.com/rewrite-tool/

This can be implemented with GS classes RBParseTreeSearcher and RBParseTreeRewriter.
- [x] 1) Add the GUI components of the Rewrite tool to Jade System Browser
- [x] 2) Implement the rewrite tool using RBParseTreeSearcher and RBParseTreeRewriter
   - [x] 2.1) Implement expressions match
   - [x] 2.2) Implement methods match
   - [x] 2.3) Add Transformations scope options (this method, this class, this hierarchy, this package, entire system??)
   - [x] 2.4) Implement AST replacement 
- [x] 3) Check which GS versions includes RBParseTreeSearcher and RBParseTreeRewriter classes (fixed with dd3089372f6e9c341d997f3de4bae74c836695ea - the commit has wrong comment that fix task 6) instead of 3))
- [x] 4) Implement a replace component to display current code and after change code. It should be a list of methods with 2 panes to display current and after code (see Dolphin RewriteChangesBrowser).
- [x] 5) Implement Apply, Apply All, Rollback and Rollback All operations
- [x] 6) Add option in JadePreferencesShell to use RBConfigurableFormatter or RBFormatter
- [x] 7)  Add options to configure Gs RBConfigurableFormatter in JadePreferencesShell
- [x] 8) Split presenters when Searching and Replacing 
- [x] 9) Remove rename argument button (there is no TransformationRule in GS)

**Comment by tukanos** [comment]
Hi @brunobuzzi ,

I wonder if the tool is finished?  Is seems like all tasks are done.  In Jade there is a tab that `RBParseTreeSearcher` and `RBParseTreeRewriter` classes are missing and thus: "The Rewrite Tool has been disabled".  How can I enable it?

I have found *Jade Rewrite tool. pac*, *Jade Rewrite tool. pax*, and *JadeRewriteTool.cls*, but there are no definitions of the `RBParseTree...` classes from above.

**Comment by brunobuzzi** [comment]
Hi,
Which version of GemStone are you running ?
This message is displayed when RBParseTreeSearcher or RBParseTreeRewriter are not present on the GemStone repository.

**Comment by tukanos** [comment]
@brunobuzzi Whoa that was fast reply! I'm running GS 3.7.2 version with Seaside extent.

**Comment by brunobuzzi** [comment]
@tukanos can you check if RBParseTreeSearcher and RBParseTreeRewriter are present in your repository ?

**Comment by tukanos** [comment]
> can you check if RBParseTreeSearcher and RBParseTreeRewriter are present in your repository ?

What do you mean by repository?  You mean the GS extent I'm running?  If yes, they are not present in the stock GS extent.

**Comment by brunobuzzi** [comment]
yes, on the GS extents. You can login with Topaz or other GS client and check if those two classes exist.

**Comment by tukanos** [comment]
> yes, on the GS extents. You can login with Topaz or other GS client and check if those two classes exist.

Yes, I did that.  Seaside or the extent0 for GS 3.7.2 does not contain these two classes.  I did not yet check the Rowan3, Rowan extents, but those are without Seaside which I need to be present.

**Comment by brunobuzzi** [comment]
maybe @jgfoster or  @dalehenrich can tell you in which package are included: RBParseTreeSearcher and RBParseTreeRewriter

**Comment by tukanos** [comment]
> maybe [@jgfoster](https://github.com/jgfoster) or [@dalehenrich](https://github.com/dalehenrich) can tell you in which package are included: RBParseTreeSearcher and RBParseTreeRewriter

Good idea.  I'll ask them.  Thank you for the tip!

**Comment by jgfoster** [comment]
Sorry, no, I don't know about the refactoring browser.

**Comment by dalehenrich** [comment]
@tukanos, I believe that the rb project is only loaded when you load tODE into a Seaside image. The class is present in the Rowan 3 image. RB is used for code formatting, but all of the RB tests should be passing in Rowan 3.

..., CC @kurtkilpela

**Comment by tukanos** [comment]
Thank you all for your answer.  I checked out the @brunobuzzi [Rewrite Tool for Jade video](https://www.youtube.com/watch?v=rfhQscqvyzY) 

@jgfoster I thought you maybe using it yourself.

@dalehenrich Ah, I see. I thought there might be a way to load it separately.  Thank you for the tip.

**Comment by dalehenrich** [comment]
>  I thought there might be a way to load it separately

It's probably worth a shot to see if it loads cleanly in a seaside extent and then see if the tests pass ...
